streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -599,47 +534,36 @@ public String toString() {
-        builder.append(indent).append("\tActive tasks:\n");
-        builder.append(active.toString(indent + "\t\t"));
-        builder.append(indent).append("\tStandby tasks:\n");
-        builder.append(standby.toString(indent + "\t\t"));
-        return builder.toString();
-    }
-
-    // this should be safe to call whether the restore consumer is assigned standby or active restoring partitions
-    // as the removal will be a no-op
-    private void removeChangelogsFromRestoreConsumer(final Collection<TopicPartition> changelogs, final boolean areStandbyPartitions) {
-        if (!changelogs.isEmpty() && areStandbyPartitions == restoreConsumerAssignedStandbys) {
-            final Set<TopicPartition> updatedAssignment = new HashSet<>(restoreConsumer.assignment());
-            updatedAssignment.removeAll(changelogs);
-            restoreConsumer.assign(updatedAssignment);
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("TaskManager\n");
+        stringBuilder.append(indent).append("\tMetadataState:\n");
+        stringBuilder.append(indent).append("\tTasks:\n");
+        for (final Task task : tasks.values()) {
+            stringBuilder.append(indent)
+                         .append("\t\t")
+                         .append(task.id())
+                         .append(" ")
+                         .append(task.state())
+                         .append(" ")
+                         .append(task.getClass().getSimpleName())
+                         .append('(').append(task.isActive() ? "active" : "standby").append(')');
         }
+        return stringBuilder.toString();
     }
 
-    private Set<TaskId> partitionsToTaskSet(final Collection<TopicPartition> partitions) {
-        final Set<TaskId> taskIds = new HashSet<>();
-        for (final TopicPartition tp : partitions) {
-            final TaskId id = partitionsToTaskId.get(tp);
-            if (id != null) {
-                taskIds.add(id);
-            } else {
-                log.error("Failed to lookup taskId for partition {}", tp);
-                throw new StreamsException("Found partition in assignment with no corresponding task");
+    // below are for testing only
+    StandbyTask standbyTask(final TopicPartition partition) {
+        for (final Task task : (Iterable<Task>) standbyTaskStream()::iterator) {
+            if (task.inputPartitions().contains(partition)) {
+                return (StandbyTask) task;
             }
         }
-        return taskIds;
-    }
-
-    // the following functions are for testing only
-    Map<TaskId, Set<TopicPartition>> assignedActiveTasks() {
-        return assignedActiveTasks;
+        return null;
     }
 
-    Map<TaskId, Set<TopicPartition>> assignedStandbyTasks() {
-        return assignedStandbyTasks;
+    // TODO: this is used from StreamThread only for a hack to collect metrics from the record collectors inside of StreamTasks
```

Comment:
```suggestion
    // TODO K9113: this is used from StreamThread only for a hack to collect metrics from the record collectors inside of StreamTasks
```

Just marking this for later follow-up.